components/feature/search/src/main/java/mozilla/components/feature/search/middleware/SearchMiddleware.kt

```diff
@@ -41,7 +47,8 @@ class SearchMiddleware(
     private val customStorage: CustomStorage = CustomSearchEngineStorage(context),
     private val bundleStorage: BundleStorage = BundledSearchEnginesStorage(context),
     private val metadataStorage: MetadataStorage = SearchMetadataStorage(context),
-    private val ioDispatcher: CoroutineContext = Dispatchers.IO
+    private val ioDispatcher: CoroutineContext = Dispatchers.IO,
+    private val applicationSearchEngines: List<SearchEngine> = emptyList(),
```

Comment:
I am very suspicious of passing in this `applicationSearchEngines` into the middleware to piggyback about on the `SetSearchEnginesAction`. I would expect the search engines in `loadSearchEngines` to need to be fetched/loaded via IO and then set, but if we don't need to do this for `applicationSearchEngines`, is there a cleaner way to go about this where we can just set it when we initialize the `SearchState()` in Fenix.

Reply:
Agreed.  Removing this from the parameters and will use `SearchState` to add the application search engine.